server/sonar-server/src/main/java/org/sonar/server/branch/pr/ws/ListAction.java

```diff
@@ -112,13 +113,12 @@ private static void addPullRequest(ProjectPullRequests.ListWsResponse.Builder re
     Optional<BranchDto> mergeBranch = Optional.ofNullable(mergeBranchesByUuid.get(branch.getMergeBranchUuid()));
     ProjectPullRequests.PullRequest.Builder builder = ProjectPullRequests.PullRequest.newBuilder();
     String branchKey = branch.getKey();
+
+    DbProjectBranches.PullRequestData pullRequestData = branch.getPullRequestData();
     builder.setId(branchKey);
-    //TODO get the branch from PULL_REQUEST_DATA
-    builder.setBranch(branchKey);
-    //TODO get the URL from PULL_REQUEST_DATA
-    builder.setUrl("https://github.com/SonarSource/sonar-core-plugins/pull/32");
-    //TODO get the branch title from PULL_REQUEST_DATA
-    builder.setTitle(branchKey);
+    builder.setBranch(pullRequestData.getBranch());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "pullRequestData" is nullable here. ([squid:S2259](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjanos%2FSONAR-10365%2Fuse-pr-id-as-key-for-pull-requests&issues=AWG38FNK42pHCPIO0Eiw&open=AWG38FNK42pHCPIO0Eiw)

[ISSUE_KEY]: AWG38FNK42pHCPIO0Eiw